src/services/oracleDatapoint.ts

```diff
@@ -0,0 +1,351 @@
+import { Pool } from "pg";
+
+import { Request, Response } from "express";
+
+interface Datapoint {
+  [address: string]: {
+    blockNumber: number;
+    blockDistance: number | null; // null if block parameter is not used
+    txHash: string;
+    txIndex: number;
+    payload: any;
+  };
+}
+
+interface Dictionary<T> {
+  [addresses: string]: T;
+}
+
+const queryMetadataOracle = `
+  SELECT *
+  FROM
+    (
+      -- newer than queried block
+      (
+        SELECT
+          txo.address as "address",
+          b.block_no AS "blockNumber",
+          CASE
+            WHEN $3::INTEGER IS NOT NULL THEN (block_no - $3)
+          END AS "blockDistance",
+          encode(tx.hash, 'hex') AS "txHash",
+          tx.block_index AS "txIndex",
+          CASE
+            WHEN $2::TEXT IS NOT NULL THEN (txm.json->$2) -- if no ticker is specified, return all tickers
+            ELSE txm.json
+          END AS "payload"
+        FROM tx_out txo
+          JOIN tx_metadata txm ON (txo.tx_id = txm.tx_id)
+          JOIN tx ON (tx.id = txm.tx_id)
+          JOIN block b ON (b.id = tx.block_id)
+        WHERE
+          txo.address = ANY ($1)
+          AND txm.key = 1968
+          AND CASE
+            WHEN $2::TEXT IS NOT NULL THEN (txm.json->$2) IS NOT NULL
+            ELSE true
+          END
+          AND CASE
+            WHEN $3::INTEGER IS NOT NULL THEN block_no >= $3 IS NOT NULL
+            ELSE true
+          END
```

Comment:
@1000101 , plz check , I think `IS NOT NULL` might be an error here, because it will return `true` for all cases when both `block_no` and `$3` are not null, e.g.
```
THEN 1 >= 10 IS NOT NULL
```
This will be true even tho the check is supposed to be `1 >= 10` and should be false